helper_scripts/test_upload.py

```diff
@@ -0,0 +1,194 @@
+    payload = json.dumps(payload)
+    r = requests.put(url, verify=self.verify, data=payload, params=params, **kwargs)
+    return r
+
+def create_user(url, api_key):
+    """
+    Create a new local galaxy user of name test-`date`,
+    with username as password.
+    """
+    gi = GalaxyInstance(url, api_key)
+    date=str(datetime.datetime.now().date())
+    username = "test-{date}".format(date=date)
+    user_email = username + "@test.com"
+    password = user_email
+    userlist = gi.users.get_users()
+    if not user_exists(username, userlist):
+        gi.users.create_local_user(username, user_email, password)
+        userlist = gi.users.get_users()
+    api_key = create_user_api_key(gi, username, userlist)
+    return api_key, user_email, password
+
+
+def user_exists(username, userlist):
+    return sum([entry["username"]==username for entry in userlist]) == 1
+
+
+def create_user_api_key(gi, username, userlist):
+    userid = [user['id'] for user in userlist if user['username'] == username][0]
+    return gi.users.create_user_apikey(userid)
+                            
+                            
+def ftp_upload_file(fqdn, user_email, password, port, tmpfile):
+    """
+    Use ftplib to upload to galaxy.
+    """
+    session = ftplib.FTP()
+    session.connect(fqdn, port)
+    session.login(user_email, password)
+    tmpfile_basename = basename(tmpfile)
+    with open(tmpfile) as file:
+        session.storbinary('STOR {fn}'.format(fn=tmpfile_basename), file)
+    session.quit()
+
+    
+def sftp_upload_file(fqdn, user_email, password, port, tmpfile):
+    """
+    Use pysftp to upload files to galaxy.
+    """
+    with pysftp.Connection(fqdn, username=user_email, password=password, port=port) as sftp:
+        sftp.put(tmpfile)
+    
+
+def successfull_upload(url, new_api_key, tmpfile):   
+    """
+    Connect as user, check if tmpfile exists in uploaded ftp files.
+    """
+    gi = GalaxyInstance(url, new_api_key)
+    files = [ True for ftp_file in gi.ftpfiles.get_ftp_files() if str(ftp_file["path"]) == basename(tmpfile)]
+    if files:
+        return True
+    else:
+        return False
+
+
+def main():
+    args = parse_args()
+    web_protocol, fqdn, ftp_protocol, port, api_key = \
+    args.protocol, args.fqdn, args.ftp_protocol, args.port, args.api_key
+    if args.http_auth_username:
+        GalaxyInstance.make_delete_request = inject_auth(make_delete_request, args.http_auth_username, args.http_auth_password)
+        GalaxyInstance.make_post_request = inject_auth(make_post_request, args.http_auth_username, args.http_auth_password)
+        GalaxyInstance.make_put_request = inject_auth(make_put_request, args.http_auth_username, args.http_auth_password)
+        GalaxyInstance.make_get_request = inject_auth(GalaxyInstance.make_get_request, args.http_auth_username, args.http_auth_password)
+    url = web_protocol + "://" + fqdn
+    new_api_key, user_email, password = create_user(url, api_key)
```

Comment:
yes... well you got the idea....
The bot is the bot. If we need to change its identity, we'll do it. But in standard days, I propose that the bot is a stable "person".